extensions/wikia/LinkSuggest/LinkSuggest.class.php

```diff
@@ -172,19 +173,42 @@ static function getLinkSuggest(WebRequest $request) {
-							( convert(binary convert(page_title using latin1) using utf8) " . $db->buildLike(strtoupper($queryLower[0]) . strtoupper($queryLower[1]) , $db->anyString() ) . " ) ) AND ";
-			} else if( strlen($queryLower) >= 1 ) {
-				$pageTitlePrefilter = "( page_title " . $db->buildLike(strtoupper($queryLower[0]) , $db->anyString() ) . " ) AND ";
+
+			// use mb_substring to get the first & second chars
+			// in case of multi-byte
+			$firstChar = mb_substr($queryLower, 0, 1);
+			$secondChar = mb_substr($queryLower, 1, 1);
+
+			if( mb_strlen( $queryLower ) >= 2 ) {
+				if ( LinkSuggest::containsMultibyteCharacters( $firstChar . $secondChar ) ) {
+					$pageTitlePrefilter = "(
+						( convert(binary convert(page_title using latin1) using utf8) LIKE convert(binary convert('" . strtoupper( $firstChar ) . strtolower( $secondChar ) . "%' using latin1) using utf8)) ) OR
+						( convert(binary convert(page_title using latin1) using utf8) LIKE convert(binary convert('" . strtoupper( $firstChar ) . strtoupper( $secondChar )  . "%' using latin1) using utf8) ) AND ";
+				} else {
+					$pageTitlePrefilter = "(
+						( convert(binary convert(page_title using latin1) using utf8) " . $db->buildLike( strtoupper( $firstChar ) . strtolower( $secondChar ) , $db->anyString() ) . " ) OR
+						( convert(binary convert(page_title using latin1) using utf8) " . $db->buildLike( strtoupper( $firstChar ) . strtoupper( $secondChar ) , $db->anyString() ) . " ) ) AND ";
+				}
+			} else if( mb_strlen($queryLower) >= 1 ) {
+				if ( LinkSuggest::containsMultibyteCharacters( $firstChar ) ) {
+					$pageTitlePrefilter = "( convert(binary convert(page_title using latin1) using utf8) LIKE convert(binary convert('" . $firstChar . "'% ) AND ";
+				} else {
+					$pageTitlePrefilter = "( convert(binary convert(page_title using latin1) using utf8) " . $db->buildLike(strtoupper( $firstChar ) , $db->anyString() ) . " ) AND ";
+				}
 			}
+
+			if ( LinkSuggest::containsMultibyteCharacters( $queryLower ) ) {
+				$pageTitleLikeClause = "convert(binary convert('{$queryLower}%' using latin1) using utf8)";
+			} else {
+				$pageTitleLikeClause = "'{$queryLower}%'";
+			}
+
 			// TODO: use $db->select helper method
 			$sql = "SELECT page_len, page_id, page_title, rd_title, page_namespace, rd_namespace, page_is_redirect
 						FROM page
 						LEFT JOIN redirect ON page_is_redirect = 1 AND page_id = rd_from
 						LEFT JOIN querycache ON qc_title = page_title AND qc_type = 'BrokenRedirects'
-						WHERE  {$pageTitlePrefilter} {$pageNamespaceClause} (convert(binary convert(page_title using latin1) using utf8) LIKE convert(binary convert('{$queryLower}%' using latin1) using utf8))
+						WHERE  {$pageTitlePrefilter} {$pageNamespaceClause} (convert(binary convert(page_title using latin1) using utf8) LIKE {$pageTitleLikeClause} )
 							AND qc_type IS NULL
 						LIMIT ".($wgLinkSuggestLimit * 3); // we fetch 3 times more results to leave out redirects to the same page
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Move this trailing comment on the previous empty line. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS139)